File: core/entities/Comment.cs
namespace core.entities
{
    public class Comment
    {
        private Guid _comment_id;
        private Guid _post_id;
        private UserAccount _user;
        private Guid _parent_id;
        private string _content;
        private DateTime _created_at;

        private int _reply_count;
        private int _like_count;

        private bool _is_liked_requester;


        public Comment(Guid comment_id, Guid post_id, UserAccount user, Guid parent_id, string content, DateTime created_at, int reply_count, int like_count, bool is_liked_requester)
        {
            this._comment_id = comment_id;
            this._post_id = post_id;
            this._user = user;
            this._parent_id = parent_id;
            this._content = content;
            this._created_at = created_at;
            this._reply_count = reply_count;
            this._like_count = like_count;
            this._is_liked_requester = is_liked_requester;
        }

        public Guid comment_id => this._comment_id;
        public Guid post_id => this._post_id;
        public UserAccount user => this._user;
        public Guid parent_id => this._parent_id;
        public string content => this._content;
        public DateTime created_at => this._created_at;
    }
}

File: client/src/utils/Comment.ts
import { IBaseUser } from "@/types/IBaseUser";

export class BaseComment {
    public post_id: string;
    public user: IBaseUser;
    public content: string;
    public created_at: Date;

    constructor(
        post_id: string,
        user: IBaseUser,
        content: string,
        created_at: Date
    ) {
        this.post_id = post_id;
        this.user = user;
        this.content = content;
        this.created_at = created_at;
    }
}

export class Comment extends BaseComment {
    public comment_id: string;
    public parent_id?: string;
    public edited: boolean;
    public edited_at?: Date;
    public reply_count: number;
    public is_hidden: boolean;
    public liked: boolean;
    public likes_count: number;

    constructor(
        post_id: string,
        user: IBaseUser,
        content: string,
        comment_id: string,
        parent_id?: string,
        created_at: Date = new Date(),
        edited = false,
        edited_at?: Date,
        reply_count = 0,
        is_hidden = false,
        liked = false,
        likes_count = 0
    ) {
        super(post_id, user, content, created_at);
        this.comment_id = comment_id;
        this.parent_id = parent_id;
        this.edited = edited;
        this.edited_at = edited_at;
        this.reply_count = reply_count;
        this.is_hidden = is_hidden;
        this.liked = liked;
        this.likes_count = likes_count;
    }
}